streams/src/test/java/org/apache/kafka/streams/integration/TaskAssignorIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+import org.junit.rules.TestName;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.apache.kafka.common.utils.Utils.mkEntry;
+import static org.apache.kafka.common.utils.Utils.mkMap;
+import static org.apache.kafka.common.utils.Utils.mkObjectProperties;
+import static org.apache.kafka.streams.integration.utils.IntegrationTestUtils.safeUniqueTestName;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.sameInstance;
+
+@Category(IntegrationTest.class)
+public class TaskAssignorIntegrationTest {
+    @ClassRule
+    public static final EmbeddedKafkaCluster CLUSTER = new EmbeddedKafkaCluster(1);
+
+    @Rule
+    public TestName testName = new TestName();
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldProperlyConfigureTheAssignor() throws NoSuchFieldException, IllegalAccessException {
+        // This test uses reflection to check and make sure that all the expected configurations really
+        // make it all the way to configure the task assignor. There's no other use case for being able
+        // to extract all these fields, so reflection is a good choice until we find that the maintenance
+        // burden is too high.
+        //
+        // Also note that this is an integration test because so many components have to come together to
+        // ensure these configurations wind up where they belong, and any number of future code changes
+        // could break this change.
+
+        final String testId = safeUniqueTestName(getClass(), testName);
+        final String appId = "appId_" + testId;
+
+        IntegrationTestUtils.cleanStateBeforeTest(CLUSTER, "input");
+
+        // Maybe I'm paranoid, but I don't want the compiler deciding that my lambdas are equal to the identity
+        // function and defeating my identity check
+        final AtomicInteger compilerDefeatingReference = new AtomicInteger(0);
+
+        // the implementation doesn't matter, we're just going to verify the reference.
+        final AssignorConfiguration.AssignmentListener configuredAssignmentListener =
+            stable -> compilerDefeatingReference.incrementAndGet();
+
+        final Properties properties = mkObjectProperties(
+            mkMap(
+                mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers()),
+                mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, appId),
+                mkEntry(StreamsConfig.NUM_STANDBY_REPLICAS_CONFIG, "5"),
+                mkEntry(StreamsConfig.ACCEPTABLE_RECOVERY_LAG_CONFIG, "6"),
+                mkEntry(StreamsConfig.MAX_WARMUP_REPLICAS_CONFIG, "7"),
+                mkEntry(StreamsConfig.PROBING_REBALANCE_INTERVAL_MS_CONFIG, "480000"),
+                mkEntry(StreamsConfig.InternalConfig.ASSIGNMENT_LISTENER, configuredAssignmentListener),
```

Comment:
Should we also validate that the task assignor gets passed through? We could even pass in a custom assignor and use that to verify the correct assignor configs got passed through.

Of course, reflection black magic-ry is just more fun 🙂

Reply:
Good idea; I'm not sure why I didn't think to do this already. With my newfound understanding of this config translation logic, I'm confident that it gets copied over right now, because it's not a registered config, but a regression test would be nice.

I'll quickly follow up with a separate PR so that I can merge this one.

Reply:
Sounds good. Thanks for the fix!